### ISSUE TITLE
It can't answer my question

### ISSUE THREAD

**Issue by Liong1976** [exoascension/vault-chat#11]
Here is what I got from this plugin.

![image](https://user-images.githubusercontent.com/88898230/225042768-01fd94a6-42e9-441c-a854-b605dedf5bbc.png)

And this is a screenshot from my console.

![image](https://user-images.githubusercontent.com/88898230/225043136-401324ea-5d2e-4d31-b94c-20db3f7cd50d.png)

I am on Windows.

**Comment by kristenbrann** [comment]
Thanks for letting us know, @Liong1976! I think there are a few things that could be happening here. 

One is the prompt engineering... under the hood we tell ChatGPT about your notes in system messages to give it context for the conversation. It can be tricky to get the best wording to set up ChatGPT for success. It would help me to understand what kind of text you have in your vault that would indicate to ChatGPT what your hobbies are? Do you have any files that explicitly refer to hobbies? Do you believe the hobbies should be able to be inferred based on other types of content? Whatever you are willing to share here will help us best understand and optimize the way we communicate with the API.

The other is that there could be an error. Based on your screenshot, I see one rate limiting error. The other messages in red are actually warnings. In order for ChatGPT to have the info it needs to answer your question, we first use embeddings to search your notes for relevant files. Then we pass the files resulting from that as context to the chat API. If the call to embeddings fails, then ChatGPT has no context from your files when we ask it your question. We are going to add some more logging and error handling for this scenario to better identify it. Without explicitly debugging, I can't be sure if this happened in your case.

**Comment by Liong1976** [comment]
Hi @kristenbrann,

Thanks for your prompt response.

Most of my notes are related to research on historical studies. On the other hand, my personal notes were written in my native language, which is a non-English language. I just replicated what you showed in the plugin's description to know if this plugin works.

As for the console, I saw a lot of red messages, but I didn't screenshot them, for they contained the data from my vault.

**Comment by cpaika** [comment]
Ok, we've released block embeddings so we're indexing your notes in chunks instead of as one giant glob.  This should have increased the accuracy of your results by a lot.  Does it seem like it's working better now if you update?  It's tough judging accuracy, it's a bit qualitative.

**Comment by Liong1976** [comment]
Thanks for the update.

When I use this plugin, will my data be uploaded to the OpenAI cloud? I don't want my data to be uploaded, for it has sensitive content, such as my passwords.

**Comment by cpaika** [comment]
Yes it will, until we have locally trained models we need to do this.  You can add an exclusion path in the plugin settings to exclude directories with sensitive content.

OpenAI's data usage policy can seen here: https://openai.com/policies/api-data-usage-policies

**Comment by Liong1976** [comment]
Thanks for the explanation.

I tried to reinstall the plugin, but it didn't work.

![image](https://user-images.githubusercontent.com/88898230/227072669-cc56c352-4f0e-4336-8ba8-eb943dd77407.png)

**Comment by cpaika** [comment]
@Liong1976 Ah sorry about that, fixed the release should work now thanks for raising the problem

**Comment by Liong1976** [comment]
Hi @cpaika,

Now the plugin works very well.  Thank you so much!

Does it upload the data to the OpenAI cloud every time I start Obsidian, or does it just add the new data?
